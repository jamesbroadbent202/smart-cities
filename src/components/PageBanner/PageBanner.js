import React from 'react';
import PropTypes from 'prop-types';
import IndicatorCard from '../Card/IndicatorCard/IndicatorCard';
import Icon from '../Icon/Icon';
import { INDICATORS } from '../../constants';
import aggregateIndicatorForCities from '../../helpers/aggregateIndicatorForCities';
import getColorVariant from '../../helpers/getColorVariant';
import style from './PageBanner.scss';

const PageBanner = (props) => {
  const backgroundColor = getColorVariant(props.colorName, '020');
  const cardHighlightColor = getColorVariant(props.colorName, '500');
  const pColor = getColorVariant(props.colorName, '900');
  const indicatorValue = props.city
    ? props.city.indices[props.indicator]
    : aggregateIndicatorForCities(
      props.indicator,
      props.cities,
    );

  return (
    <div style={{ backgroundColor }}>
      <div className={style.container}>
        <div className={style.textAndIndicator}>
          <div className={style.titleText}>
            <h1 className={style.title}>{props.title}</h1>

            <p className={style.description}>{props.description}</p>
          </div>

          <IndicatorCard
            size="large"
            className={style.card}
            color={cardHighlightColor}
            colorName={props.colorName}
            indicator={props.indicator}
            value={indicatorValue}
            isCategoryPage={props.isCategoryPage}
          />
        </div>

        {!props.isContextPage && (
          <div className={style.indicatorTypeMarkWrapper}>
            <Icon
              className={style.indicatorTypeMark}
              color={pColor}
              icon={props.isCategoryPage ? 'indicatorTypeMarkSolid' : 'indicatorTypeMarkBorder'}
              size={22}
            />

            Performance indicators
          </div>
        )}
      </div>
    </div>
  );
};

const cityPropShape = {
  id: PropTypes.string.isRequired,
  indices: PropTypes.object.isRequired,
  name: PropTypes.string.isRequired,
};

PageBanner.propTypes = {
  cities: PropTypes.arrayOf(PropTypes.shape(cityPropShape)),
  city: PropTypes.shape(cityPropShape),
  colorName: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  indicator: PropTypes.oneOf(Object.keys(INDICATORS)).isRequired,
  title: PropTypes.string.isRequired,
  isCategoryPage: PropTypes.bool,
  isContextPage: PropTypes.bool,
};

export default PageBanner;
