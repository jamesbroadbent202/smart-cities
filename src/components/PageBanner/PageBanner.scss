@import '~style/variables';

.container {
  @include container;

  padding-top: 16px;
  padding-bottom: 16px;

  @include for-tablet-portrait-up {
    padding-top: 40px;
    padding-bottom: 32px;
  }
}

.textAndIndicator {
  display: flex;
  flex-direction: column;
  margin-bottom: 16px;

  @include for-tablet-portrait-up {
    flex-direction: row;
    margin-bottom: 32px;
  }
}

.titleText {
  margin-bottom: 32px;

  @include for-tablet-portrait-up {
    margin-bottom: 0;
    margin-right: 32px;
  }
}

.title {
  @include type-h1;

  margin-bottom: 16px;
}

.description {
  @include type-intro-text;
}

.card {
  width: 100%;

  @include for-tablet-portrait-up {
    flex: none;
    width: 300px; // TODO (davidg): should allow for the widest value
    margin-left: auto; // push to the right
  }
}

.indicatorTypeMarkWrapper {
  display: flex;
  align-items: center;
}

.indicatorTypeMark {
  margin-right: 8px;
}
