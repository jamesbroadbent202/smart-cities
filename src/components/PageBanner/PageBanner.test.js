import React from 'react';
import { shallow } from 'enzyme';
import PageBanner from './PageBanner';

jest.mock('../Card/IndicatorCard/IndicatorCard', () => 'IndicatorCard');
jest.mock('../PageLegend/PageLegend', () => 'PageLegend');

const defaultProps = {
  colorName: 'jobs',
  description: 'This is the mock category',
  indicator: 'population',
  title: 'Banner title',
  cities: [
    {
      id: 'perth',
      name: 'Perth',
      indices: {
        population: 77,
      }
    },
    {
      id: 'sydney',
      name: 'Sydney',
      indices: {
        population: 23,
      }
    },
  ],
};

it('should match Snapshot', () => {
  const component = shallow(
    <PageBanner {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should set the background color', () => {
  const component = shallow(
    <PageBanner {...defaultProps} />
  );

  const componentStyle = component.find('div').at(0).prop('style');
  // note, here the color is the string 'JOBS_020' because of the way
  // styles are imported during tests, but in reality
  // this will be the hex color
  expect(componentStyle.backgroundColor).toBe('JOBS_020');
});

it('should get the min and max values', () => {
  const component = shallow(
    <PageBanner {...defaultProps} />
  );

  expect(component.find('IndicatorCard').prop('value')).toEqual([23, 77]);
});

it('should show data for a single city', () => {
  const component = shallow(
    <PageBanner
      {...defaultProps}
      cities={undefined}
      city={{
        id: 'perth',
        name: 'Perth',
        indices: {
          population: 77,
        },
      }}
    />
  );

  expect(component.find('IndicatorCard').prop('value')).toBe(77);
});
