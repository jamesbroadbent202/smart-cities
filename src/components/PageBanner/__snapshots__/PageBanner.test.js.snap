// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div style={{...}}>
  <div className=\\"container\\">
    <div className=\\"textAndIndicator\\">
      <div className=\\"titleText\\">
        <h1 className=\\"title\\">
          Banner title
        </h1>
        <p className=\\"description\\">
          This is the mock category
        </p>
      </div>
      <IndicatorCard size=\\"large\\" className=\\"card\\" color=\\"JOBS_500\\" colorName=\\"jobs\\" indicator=\\"population\\" value={100} isCategoryPage={[undefined]} />
    </div>
    <div className=\\"indicatorTypeMarkWrapper\\">
      <Icon className=\\"indicatorTypeMark\\" color=\\"JOBS_900\\" icon=\\"indicatorTypeMarkBorder\\" size={22} />
      Performance indicators
    </div>
  </div>
</div>"
`;
